crates/ruff_python_formatter/src/expression/expr_attribute.rs

```diff
@@ -35,105 +35,117 @@ impl FormatNodeRule<ExprAttribute> for FormatExprAttribute {
 
         let call_chain_layout = self.call_chain_layout.apply_in_node(item, f);
 
-        let needs_parentheses = matches!(
-            value.as_ref(),
-            Expr::Constant(ExprConstant {
-                value: Constant::Int(_) | Constant::Float(_),
-                ..
-            })
-        );
+        let format_inner = format_with(|f: &mut PyFormatter| {
```

Comment:
(This is purely indenting the existing body in a `format_with`. No changes in the indented body.)